lightning/src/chain/keysinterface.rs

```diff
@@ -195,9 +195,12 @@ impl Readable for SpendableOutputDescriptor {
 // TODO: We should remove Clone by instead requesting a new ChannelKeys copy when we create
 // ChannelMonitors instead of expecting to clone the one out of the Channel into the monitors.
 pub trait ChannelKeys : Send+Clone {
-	/// Gets the commitment seed for a specific commitment number
+	/// Gets the per-commitment point for a specific commitment number
 	/// Note that the commitment number starts at (1 << 48) - 1 and counts backwards
-	fn commitment_secret(&self, idx: u64) -> [u8; 32];
+	fn get_per_commitment_point<T: secp256k1::Signing + secp256k1::Verification>(&self, idx: u64, secp_ctx: &Secp256k1<T>) -> PublicKey;
+	/// Gets the commitment seed for a specific commitment number, thereby revoking the commitment
+	/// Note that the commitment number starts at (1 << 48) - 1 and counts backwards
+	fn revoke_commitment(&self, idx: u64) -> [u8; 32];
```

Comment:
Hm, given this function is also used to get the secret for channel_reestablish, not just revocation, seems the original name `commitment_secret` (or maybe `get_commitment_secret` or so) might be more appropriate?

Reply:
Or maybe get_revoke_commitment_secret or so, to indicate that its both revoking and that it may be called twice for the same commitment in the reestablish case - either way, worth nothing that this may happen in the docs.

Reply:
Using Matt's name, which I think attracts attention to the fact that this is not a pure function, but also has some policy enforcement side effects.

Reply:
I'm not quite sure I follow why "revoke" is in the name. Isn't that dependent on the context in which the method is called?

I'd also rather not use the `get_` prefix for accessors as that isn't the convention.

Reply:
Actually, after this PR getting the secret is *only* used for RevokeAndACK.  Let me know if I'm missing something.

Reply:
Understood. But "revoke" describes how the caller (i.e., `Channel`) is using this function. The caller shouldn't need clarification (by including "revoke" in the name) on how it intends to use the result of a function that it calls.

Separately, I see the value of documenting this behavior for the implementor of the trait. In that respect, higher-level documentation on the trait itself would be most valuable rather than spreading across function-level documentation. But this is more of a critique of how the overall documentation of this trait is currently organized than the specific change in this PR.

Regarding my earlier comment on the `get_` prefix, I see your point that it's not a pure function. Though, I think using "get" doesn't imply that it's not either given that prefix is typically used for accessor methods in some languages. Given the in-memory implementation delegates to `build_commitment_secret`, perhaps "build" is more appropriate prefix.

Reply:
@jkczyz, you also came up with `release_commitment_secret`.  I like that one quite a bit, since it has an action verb and seems less cumbersome.

Reply:
Yes, after some offline discussion @devrandom convinced me that -- from the point of view of the signer -- the commitment has been revoked. After some pondering, I thought `release_commitment_secret` seemed suitable since it signals the secret is in the wild but the actual revocation happens at a higher level of abstraction.

Reply:
done